react-native-async-storage/async-storage
# feat(expo): add an Expo config plugin to simplify enabling Next Storage implementation

## Summary



Adds a config plugin to enable [Next storage implementation](https://react-native-async-storage.github.io/async-storage/docs/advanced/next/).

Fixes #750

> [!NOTE]
> As of Expo v51, this will cause issues with the KSP version lookup so users will also need to use Expo's [`BuildProperties` plugin](https://docs.expo.dev/versions/latest/sdk/build-properties/#pluginconfigtypeandroid) to set `kotlinVersion` to `1.9.23`

## Test Plan



1. First, build the plugin:

```bash
yarn turbo run build --filter @react-native-async-storage/expo-plugin
```

2. Link the newly-built package into your Expo project (`bun/npm/yarn link`)

3. Add the config plugin to your Expo project's `app.json`:

```json
{
  "expo": {
    "plugins": [
      [
        "@react-native-async-storage/expo-plugin",
        {
          "android": {
            "nextStorage": {
              "enabled": false
            }
          }
        }
      ]
    ]
  }
}
```

4. Run prebuild in your Expo project:

```bash
npx expo prebuild -p android --clean
```

5. Check `android/gradle.properties` to confirm `AsyncStorage_useNextStorage=true` is there
